### DISCUSSION TITLE
http-server

### DISCUSSION THREAD

**Discussion by boomzero** [XMOJ-Script-dev/XMOJ-Script#349] [Show and tell]
https://github.com/boomzero/http_server

http-server is a static file server written in a single C++ file using sockets

It can serve xmoj-script's homepage:
<img width="1436" alt="image" src="https://github.com/XMOJ-Script-dev/XMOJ-Script/assets/85378277/1bbfdd18-93ab-4457-a60f-8a95927ebc4a">
It is very fast, and uses 4 threads.

**Comment by boomzero** [comment]
@langningchen 手写socket

**Comment by langningchen** [comment]
@boomzero 和我一样真厉害
